### PR TITLE
protocol/tx: add benchmarks for entryID

### PR DESCRIPTION
**Pull request by erykwalder**
Adds benchmarks for entryID across several types of entries.

### CODE REVIEW

File: protocol/tx/entry_test.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import (
+	"chain/protocol/bc"
+	"testing"
+	"time"
+)
+
+func BenchmarkEntryIDIssuance(b *testing.B) {
+	entry := newIssuance(bc.Hash{}, bc.AssetAmount{}, bc.Hash{}, 0)
+	for i := 0; i < b.N; i++ {
+		_ = entryID(entry)
```
Comment (kr) [inline protocol/tx/entry_test.go]: Is the `_ =` there for a specific purpose?

File: protocol/tx/entry_test.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		Value:    bc.AssetAmount{},
+		Position: 1,
+	}, bc.Hash{}, 1)
+	for i := 0; i < b.N; i++ {
+		_ = entryID(entry)
+	}
+}
+
+func BenchmarkEntryIDSpend(b *testing.B) {
+	entry := newSpend(bc.Hash{}, bc.Hash{}, 0)
```
Comment (kr) [inline protocol/tx/entry_test.go]: Do you want to consider a subbenchmark?

```go
func BenchmarkEntryID(b *testing.B) {
	cases := []entry{
		newIssuance(bc.Hash{}, bc.AssetAmount{}, bc.Hash{}, 0),
		newHeader(1, []bc.Hash{{}}, bc.Hash{}, uint64(time.Now().Unix()), uint64(time.Now().Unix())),
		// ...
		newSpend(bc.Hash{}, bc.Hash{}, 0),
	}

	for _, e := range cases {
		name := reflect.TypeOf(e).Elem().Name()
		b.Run(name, func(b *testing.B) {
			for i := 0; i < b.N; i++ {
				entryID(e)
			}
		})
	}
}
```

Comment (kr) [inline protocol/tx/entry_test.go]: Oops forgot to link to the godoc https://godoc.org/testing#B.Run.

**Comment by erykwalder** [comment]
updated, PTAL

**Comment by kr** [comment]
LGTM

**Pull request merged**